src/php/Command/FormatCommand.php

```diff
@@ -4,23 +4,75 @@
 
 namespace Phel\Command;
 
+use Phel\Command\Shared\CommandIoInterface;
 use Phel\Formatter\FormatterInterface;
 
 final class FormatCommand
 {
     public const COMMAND_NAME = 'fmt';
 
+    private const PHEL_EXTENSION = 'phel';
+
     private FormatterInterface $formatter;
+    private CommandIoInterface $io;
 
-    public function __construct(FormatterInterface $formatter)
-    {
+    /** @var list<string> */
+    private array $filePaths = [];
+
+    public function __construct(
+        FormatterInterface $formatter,
+        CommandIoInterface $io
+    ) {
         $this->formatter = $formatter;
+        $this->io = $io;
+    }
+
+    public function run(array $paths): void
+    {
+        foreach ($paths as $fileOrPath) {
```

Comment:
Wouldn't it be better to first collect all files? Then you can filter out duplicates. For example if i run
```
phel fmt src/file-1.phel src/file-1.phel
```
This will format the file twice and not once.

Reply:
I improved this part filtering the inputs (by unique values) and also saved in a variable the already parsed files to avoid run the fmt twice on the same files at any chance.